src/tools/clippy/clippy_lints/src/attrs.rs

```diff
@@ -342,6 +340,42 @@ declare_clippy_lint! {
     "ensures that all `allow` and `expect` attributes have a reason"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for usage of the `#[allow]` attribute and suggests replacing it with
+    /// the `#[expect]` (See [RFC 2383](https://rust-lang.github.io/rfcs/2383-lint-reasons.html))
+    ///
+    /// This lint only warns outer attributes (`#[allow]`), as inner attributes
+    /// (`#![allow]`) are usually used to enable or disable lints on a global scale.
+    ///
+    /// ### Why is this bad?
+    /// `#[expect]` attributes suppress the lint emission, but emit a warning, if
+    /// the expectation is unfulfilled. This can be useful to be notified when the
+    /// lint is no longer triggered.
+    ///
+    /// ### Example
+    /// ```rust,ignore
+    /// #[allow(unused_mut)]
+    /// fn foo() -> usize {
+    ///    let mut a = Vec::new();
+    ///    a.len()
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```rust,ignore
+    /// #![cfg_attr(bootstrap, feature(lint_reasons))]
```

Comment:
Shouldn't these be commented out? Or removed if what oli said about ui tests applies to doctests, idrk

Reply:
Maybe? I believe, I tried that in the previous stabilization PR and got an error at that time. The docs are built for the beta and stable channel, to I would guess that they are needed. I'm trying to run the rest right now (Rustc takes so long to compile :sob: Even on the dev machines from rust-lang)

Reply:
It should be fine to remove them. Clippy is a stage1 tool